addons/mail/models/mail_notification.py

```diff
@@ -111,12 +111,17 @@ def format_failure_reason(self):
     # DISCUSS
     # ------------------------------------------------------------
 
+    def _is_display_notification(self):
+        """ Returns whether the notification should be displayed or not in the web client """
+        self.ensure_one()
```

Comment:
I would keep this code in `_filtered_for_web_client` because it's only used there

Reply:
Done, I have moved (and renamed) the function as a local function of _filtered_for_web_client because it is more readable than a lambda given the number of conditions